README.md

```diff
@@ -996,6 +996,62 @@ func main() {
 }
 ```
 
+##### Perl using each
+
+```perl
+my %hash = ( key_1 => 'foo', key_2 => 'bar', );
+for my $key, $value ( each %hash ) {
+    printf( "key: %s value: %s\n", $key, $value );
+}
+
+```
+
+##### Go using only primary and secondary return values
+
+```go
+package main
+
+import (
+	"fmt"
+)
+
+func main() {
+	myMap := map[string]string{"key1": "foo", "key2": "bar"}
+
+	for _, v := range myMap {
+		fmt.Printf("key: %s value: %s\n", k, v)
+	}
+}
+```
+
+##### Perl using values
+
+```perl
+my %hash = ( key_1 => 'foo', key_2 => 'bar', );
+for my $key, $value ( values %hash ) {
+    printf( "key: %s value: %s\n", $key, $value );
+}
```

Comment:
I think there may be a copy/paste error here, since `values` should not return a key/value pair.

Reply:
Thanks, I really messed up the examples